packages/salesforce/src/Adaptor.js

```diff
@@ -659,56 +659,103 @@ function setApiVersion(apiVersion) {
 
   return version;
 }
+
 /**
- * Creates a connection.
+ * Establish a connection to Salesforce using username and password.
  * @example
- * createConnection(state)
+ * createBasicAuthConnection(state)
  * @function
  * @param {State} state - Runtime state.
  * @returns {State}
  */
-function createConnection(state) {
-  const { loginUrl, apiVersion } = state.configuration;
+async function createBasicAuthConnection(state) {
+  console.log('Attempting to connect with Basic Auth');
+
+  const { loginUrl, apiVersion, username, password, securityToken } =
+    state.configuration;
 
-  if (!loginUrl) {
-    throw new Error('loginUrl missing from configuration.');
+  if (!loginUrl || !username || !password) {
+    throw new Error(
+      'Incomplete configuration for basic authentication. Provide loginUrl, username, password, and securityToken if required.'
```

Comment:
The `if required` bit is a little confusing - which keys are required? We don't know.  The user probably does.

I would be tempted to just let the login fail and report the error message from salesforce back to the user. Assuming we get a good error.

The argument against this is that if our users regularly are omitting certain information (like the login url, which seems like an obvious thing to forget), we should warn. I presume that's why we did this for loginUrl. I don't think it's useful for username and password though.

Reply:
@josephjclark 
The required key are `username, password and loginUrl`. I used the jsforce language `if required` because not all salesforce instance requires `securityToken` - https://jsforce.github.io/document/#username-and-password-login

If a user does not provide those required properties, the salesforce error log will say
```
INVALID_LOGIN: Invalid username, password, security token; or user locked out.
```
To me that error suggests i have entered the wrong information rather than telling me i need to fill in the required values in my `state.configuration` 

So if it's a warning i think it should consider both required values rather than looking for `loginUrl` only

Reply:
Remove the if statement, salesforce will return `INVALID_LOGIN: Invalid username, password, security token; or user locked out.` If required fields are not specified in `state.configuration`